File: src/1773_count-items-matching-a-rule.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        rule_key_idx = {'type': 0, 'color': 1, 'name': 2}
        count = 0

        for item in items:
            count += int(item[rule_key_idx[ruleKey]] == ruleValue)

        return count

File: Elixir/1773_count-items-matching-a-rule.ex
defmodule Solution do
  @key_idx_mapper %{"type" => 0, "color" => 1, "name" => 2}

  @spec count_matches(items :: [[String.t()]], rule_key :: String.t(), rule_value :: String.t()) ::
          integer
  def count_matches(items, rule_key, rule_value) do
    Enum.count(items, fn item ->
      Enum.at(item, @key_idx_mapper[rule_key]) == rule_value
    end)
  end
end

# Slightly fancier solution

# defmodule Solution do
#   @spec count_matches(items :: [[String.t()]], rule_key :: String.t(), rule_value :: String.t()) ::
#           integer
#   def count_matches(items, rule_key, rule_value) do
#     Enum.count(items, fn item ->
#       case {item, rule_key} do
#         {[^rule_value, _, _], "type"} -> true
#         {[_, ^rule_value, _], "color"} -> true
#         {[_, _, ^rule_value], "name"} -> true
#         _ -> false
#       end
#     end)
#   end
# end
